### PR TITLE
JAMES-3775 Quick rework ClamAVScan mailet

### PR DESCRIPTION
**Pull request by quantranhong1999**
Following the James <-> Rspamd <-> ClamAV integration, I would like to do a quick rework for the `ClamAVScan` mailet so we can have more deployment choices ( James <-> ClamAV).

### CODE REVIEW

**Review by chibenwa — COMMENTED** [review]
How about moving ClamAV related stuff in a dedicated `third-party/clamav` maven module as a separate extension?

**Comment by vttranlina** [comment]
read it

**Review by Arsnael — APPROVED** [review]

**Comment by quantranhong1999** [comment]
Rebased on master.
> How about moving ClamAV related stuff in a dedicated third-party/clamav maven module as a separate extension?

https://github.com/apache/james-project/pull/1134/commits/0a9239e8e13500990ba24b7a3483972355f85385 solves.

File: third-party/clamav/pom.xml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (chibenwa) [inline third-party/clamav/pom.xml:2]: License?

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
```
Failed

org.apache.james.clamav.ClamAVScanTest.org.apache.james.clamav.ClamAVScanTest

12:25:24.094 [ERROR] ?.105] - Could not start container
org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (localhost ports: [49623] should be listening)

12:25:24.129 [ERROR] ?.105] - Log output from the failed container:
Starting ClamAV
LibClamAV Warning: **************************************************
LibClamAV Warning: ***  The virus database is older than 7 days!  ***
LibClamAV Warning: ***   Please update it as soon as possible.    ***
LibClamAV Warning: **************************************************
```

On my checklist: ensure that the clamAV virus database do not get refreshed every time the clamAV container starts...

File: mailet/standard/pom.xml [hunk 2]
```diff
@@ -46,6 +46,10 @@
             <artifactId>apache-mailet-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-core</artifactId>
+        </dependency>
         <dependency>
```
Comment (chibenwa) [inline mailet/standard/pom.xml:53]: This dependency is no longer needed

File: third-party/clamav/src/test/java/org/apache/james/clamav/DockerClamAV.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+public class DockerClamAV {
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("clamav/clamav");
+    private static final String DEFAULT_TAG = "0.105";
+    private static final int DEFAULT_PORT = 3310;
+
+    private final GenericContainer<?> container;
+
+    public DockerClamAV() {
+        this.container = new GenericContainer<>(DEFAULT_IMAGE_NAME.withTag(DEFAULT_TAG))
+            .withExposedPorts(DEFAULT_PORT);
```
Comment (chibenwa) [inline third-party/clamav/src/test/java/org/apache/james/clamav/DockerClamAV.java:34]: Please add https://github.com/apache/james-project/pull/1139/commits/3822c62641a444c12995f18f77bb7c62095898cd

```
            .withEnv("CLAMAV_NO_FRESHCLAMD", "true")
            .withEnv("CLAMAV_NO_MILTERD", "true")
```

**Review by chibenwa — COMMENTED** [review]
Also squash fixups?

File: third-party/clamav/src/test/java/org/apache/james/clamav/DockerClamAV.java (hunk 3 above)
Comment (quantranhong1999) [reply third-party/clamav/src/test/java/org/apache/james/clamav/DockerClamAV.java]: https://github.com/apache/james-project/pull/1134/commits/71fde51d646439db4d12e312ec0504b5973b24d6

**Comment by chibenwa** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1134/13/testReport/junit/org.apache.james.clamav/ClamAVScanTest/org_apache_james_clamav_ClamAVScanTest/

Some ClamAV related failures (container did not startup).

Can we at least increase the timeout to something large?

**Comment by Arsnael** [comment]
Can we merge this or not then?

**Comment by chibenwa** [comment]
> Can we merge this or not then?

Let's go, if it annoys us we will tag it as unstable.

**Pull request merged**